sdk/samples/iot/pnp/pnp_thermostat_component.c

```diff
@@ -171,154 +191,340 @@ az_result pnp_thermostat_init(
-    pnp_thermostat_component* ref_thermostat_component,
-    az_json_token const* property_name,
-    az_json_reader const* property_value,
+// pnp_thermostat_build_response_payload builds the JSON payload that acknowledges a
+// targetTemperature request
+static void pnp_thermostat_build_response_payload(
+    az_iot_hub_client const* hub_client,
+    pnp_thermostat_component* thermostat_component,
+    double targetTemperature,
     int32_t version,
     az_span payload,
     az_span* out_payload)
 {
-  double parsed_property_value = 0;
+  const char* const property_log = "Failed to create property with status payload";
+
+  az_json_writer jw;
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(
+      az_json_writer_init(&jw, payload, NULL), "Failed to initialize the json writer");
+
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(az_json_writer_append_begin_object(&jw), property_log);
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(
+      az_iot_hub_client_properties_builder_begin_component(
+          hub_client, &jw, thermostat_component->component_name),
+      property_log);
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(
+      az_iot_hub_client_properties_builder_begin_response_status(
+          hub_client,
+          &jw,
+          property_desired_temperature_property_name,
+          (int32_t)AZ_IOT_STATUS_OK,
+          version,
+          property_response_success),
+      property_log);
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(
+      az_json_writer_append_double(&jw, targetTemperature, DOUBLE_DECIMAL_PLACE_DIGITS),
+      property_log);
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(
+      az_iot_hub_client_properties_builder_end_response_status(hub_client, &jw), property_log);
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(
+      az_iot_hub_client_properties_builder_end_component(hub_client, &jw), property_log);
+
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(az_json_writer_append_end_object(&jw), property_log);
 
-  if (!az_json_token_is_text_equal(property_name, twin_desired_temperature_property_name))
+  *out_payload = az_json_writer_get_bytes_used_in_destination(&jw);
+}
+
+void pnp_thermostat_process_property_update(
+    pnp_thermostat_component* thermostat_component,
+    az_iot_hub_client const* hub_client,
+    MQTTClient mqtt_client,
+    az_json_reader* property_name_and_value,
+    int32_t version)
+{
+  char const* const log_message = "Failed to process property update";
+  double targetTemperature = 0;
+
+  pnp_mqtt_message publish_message;
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(
+      pnp_mqtt_message_init(&publish_message), "Failed to initialize pnp_mqtt_message");
+
+  if (!az_json_token_is_text_equal(
+          &property_name_and_value->token, property_desired_temperature_property_name))
   {
-    return false;
+    // The requested property is not supported by this model.  We do NOT send any error
+    // response back to IoT Hub in this instance.  The function is responsible however
+    // for advancing the property JSON reader to set up the next property enumeration by the caller.
```

Comment:
Just double checking here.....this is after we've already determined that a property is supposedly associated with this component in the document right? This isn't a "check if associated with this component and update" call

Reply:
Correct - we see that the component=="thermostat1" or component=="thermostat2" in the caller, which dispatches to this routine to process.  And this routine then goes and processes the underlying properties, or in this block's case detects there's no property set.